src/components/VmStatus/getVmStatus.js

```diff
@@ -156,17 +126,132 @@ const isWaitingForVmi = vm => {
   return NOT_HANDLED;
 };
 
-export const getVmStatusDetail = (vm, launcherPod, importerPods, migration) =>
-  isBeingMigrated(vm, migration) || // must be precceding isRunning() since vm.status.ready is true for a migrating VM
+const findCdiPod = (volume, cdiPods = []) =>
+  cdiPods.find(pod => {
+    const podName = getName(pod);
+    const dataVolumeName = volume.dataVolume.name;
+    // upload dataVolume pod
+    if (podName === `cdi-upload-${dataVolumeName}`) {
+      return true;
+    }
+    // import dataVolume pod
+    const importerPodName = `importer-${dataVolumeName}-`;
+    if (podName.startsWith(importerPodName) && podName.length === importerPodName.length + 5) {
+      return true;
+    }
+    // clone dataVolume pod
+    if (getLabels(pod)['cdi.kubevirt.io/storage.clone.cloneUniqeId'] === `${dataVolumeName}-target-pod`) {
+      return true;
+    }
+    return false;
+  });
+
+const getPodError = cdiPod => {
+  if (cdiPod) {
+    if (!isSchedulable(cdiPod)) {
+      return 'Pod scheduling failed.';
+    }
+
+    const failingContainer = getFailingContainerStatus(cdiPod);
+    if (failingContainer) {
+      return getContainerStatusReason(failingContainer);
+    }
+  }
+  return NOT_HANDLED;
+};
+
+export const isPreparingDisks = (vm, cdiPods, dataVolumes) => {
+  const diskStatuses = [];
+  if (dataVolumes) {
```

Comment:
I would prefer an early return here to stop the indentation

Reply:
added early return and also another one a few lines later